fastembed/text/text_embedding.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Union, Iterable, List, Dict, Any, Type
+
+import numpy as np
+
+from fastembed.text.e5_onnx_embedding import E5OnnxEmbedding
+from fastembed.text.jina_onnx_embedding import JinaOnnxEmbedding
+from fastembed.text.onnx_embedding import OnnxTextEmbedding
+from fastembed.text.text_embedding_base import TextEmbeddingBase
+
+
+class TextEmbedding(TextEmbeddingBase):
+    EMBEDDINGS_REGISTRY: List[Type[TextEmbeddingBase]] = [
+        OnnxTextEmbedding,
+        E5OnnxEmbedding,
+        JinaOnnxEmbedding,
+    ]
+
+    @classmethod
```

Comment:
How do we handle the case of models which we use for both dense and sparse? E.g. [BGE-M3](https://huggingface.co/BAAI/bge-m3)

Reply:
we would need to implement both TextEmbedding and SparseEmbedding(tba) classes for those models. I am strictly against changing the function output format depending on the model name